lib/std/hash/streaming_cache.zig

```diff
@@ -0,0 +1,34 @@
+const std = @import("std");
+
+/// Abstracts buffer management for unaligned input blocks for hash and crypto functions.
+pub fn StreamingCache(comptime round_length: usize) type {
```

Comment:
Should rename. This isn't really a cache, more just temporary storage for streaming.

Reply:
StreamingBuffer? Afaiu, you are using a temporary buffer for streaming.

Reply:
I agree that it is very confusing. How about `HashBuffer` or something like that?

That being said, maybe there's an easier and more efficient approach: simply add an `hashSlice(comptime N: usize, m: [N]u8)` function, since small keys generally have a compile-known size.

The optimizer may then be able to removes all the unnecessary steps for a given size.

Reply:
I tried that approach, and it seems to be working pretty well.

```
xxhash32 before: 5049 MiB/s [1ffbad07bda51000]
xxhash32 after :  14895 MiB/s
```

Just by adding a `hashSlice()` function like this:

```zig
    pub fn hashSlice(comptime T: usize, seed: u32, input: *const [T]u8) u32 {
        var hasher = @call(.always_inline, init, .{seed});
        @call(.always_inline, update, .{ &hasher, input });
        return @call(.always_inline, final, .{&hasher});
    }
```